libsolidity/codegen/ir/IRGenerationContext.cpp

```diff
@@ -180,7 +180,10 @@ ABIFunctions IRGenerationContext::abiFunctions()
 
 uint64_t IRGenerationContext::internalFunctionID(FunctionDefinition const& _function, bool _requirePresent)
 {
-	auto [iterator, inserted] = m_functionIDs.try_emplace(_function.id(), m_functionIDs.size() + 1);
+	auto functionID = _function.annotation().internalFunctionID;
+	solAssert(functionID.has_value(), "");
+
+	auto [iterator, inserted] = m_functionIDs.try_emplace(_function.id(), functionID.value());
 	if (_requirePresent)
 			solAssert(!inserted, "");
 	return iterator->second;
```

Comment:
```suggestion
	return *functionID;
```

We don't need ``m_functionIDs`` or any of this logic anymore, do we? Asserting that ``functionID.has_value()`` is just enough - then we can just return it.

Reply:
To be clear: if I'm not missing a reason to keep it, just remove the ``m_functionIDs`` member entirely.

Reply:
I kept it because of the `copyFunctionIDsFrom` function, which is used by the `IRGenerator::resetContext` to copy the internal function IDs from the constructor code to deployed code. But now you make me remember that you said weeks ago that the internal IDs are assumed to be the same during creation and runtime, so I can remove this indeed.